dev-tools/e2e/build.sh

```diff
@@ -0,0 +1,39 @@
+#!/bin/bash
+
+# This script builds an image from the elastic-agent image
+# with a locally built fleet-server binary injected. Additional
+# flags (e.g. -t <name>) will be passed to `docker build`.
```

Comment:
This file is the same as `dev-tools/cloud/docker/build.sh` but uses the base image version from `dev-tools/integration/.dev` and does not push the image.

I want to make  tests that runs this docker container so we can verify fleet-server bootstrapping works